crates/ruff_python_ast/ast.toml

```diff
@@ -113,6 +106,35 @@ ExprList = {}
 ExprTuple = {}
 ExprSlice = {}
 ExprIpyEscapeCommand = {}
+ExprBoolOp = { fields = [
+    { name = "op", type = "BoolOp" },
+    { name = "values", type = "Expr", seq = true }
+]}
```

Comment:
> Is the AST definition easy to understand and maintain?

Given this concern, this syntax does start to look a bit unwieldy.

> Is it easy to parse in our code generator?

But on the other hand, continuing to use TOML still makes it easier to iterate on the code generator itself.

So given all of this, I'd still lean towards using this TOML syntax.  We can always revisit the syntax later if we feel there's something that e.g. ungrammar would give us that the TOML doesn't.